File: jar-common/src/main/java/com/beancounter/common/utils/CurrencyUtils.java
package com.beancounter.common.utils;

import com.beancounter.common.model.Currency;
import com.beancounter.common.model.IsoCurrencyPair;
import java.math.BigDecimal;
import lombok.experimental.UtilityClass;

@UtilityClass
public class CurrencyUtils {
  public Currency getCurrency(String isoCode) {
    if ( isoCode == null) {
      return null;
    }
    return Currency.builder().code(isoCode).build();
  }

  public IsoCurrencyPair getCurrencyPair(BigDecimal rate, Currency from, Currency to) {
    IsoCurrencyPair isoCurrencyPair = null;
    if (from == null || to == null) {
      return null;
    }

    if (rate == null && !from.getCode().equalsIgnoreCase(to.getCode())) {
      isoCurrencyPair = IsoCurrencyPair.builder()
          .from(from.getCode())
          .to(to.getCode())
          .build();
    }
    return isoCurrencyPair;
  }
}

File: jar-common/src/main/java/com/beancounter/common/utils/CurrencyUtils.kt
package com.beancounter.common.utils

import com.beancounter.common.model.Currency
import com.beancounter.common.model.IsoCurrencyPair
import java.math.BigDecimal

class CurrencyUtils {
    fun getCurrency(isoCode: String): Currency {
        return Currency(isoCode, null, null)
    }

    fun getCurrencyPair(rate: BigDecimal?, from: Currency, to: Currency): IsoCurrencyPair? {
        if (rate == null && !from.code.equals(to.code, ignoreCase = true)) {
            return IsoCurrencyPair(from = from.code, to = to.code)
        }
        return null
    }
}